src/Web/Router/Route.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Web\Router;
+
+use App\Web\Request\Request;
+use App\Web\Response\Response;
+
+class Route
+{
+    private array $handlers = [];
+
+    public function __construct(
+        private readonly string $pattern,
+        private readonly Response $notAllowed,
+        array $handlers
+    ) {
+        foreach ($handlers as $key => $handler) {
+            $this->handlers[strtoupper($key)] = $handler;
+        }
```

Comment:
This will fail when index (key) will be integer

<img width="1360" alt="image" src="https://github.com/user-attachments/assets/d318dc03-c8eb-47a2-8924-2688868bcc89">

I would suggest to delete this, and ensure uppercase keys in handle method using `array_change_key_case`, or if you wish you could replace current solution with `array_change_key_case` https://www.php.net/manual/en/function.array-change-key-case.php